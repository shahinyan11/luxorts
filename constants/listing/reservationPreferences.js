export const LISTING_PRIOR_NOTIFIED_TYPE = {
  SAME_DAY: 'same_day',
  ONE_DAY: '1_day',
  TWO_DAYS: '2_days',
  THREE_DAYS: '3_days',
  SEVEN_DAYS: '7_days',
};

export const LISTING_PRIOR_NOTIFIED_OPTIONS = [
  { value: LISTING_PRIOR_NOTIFIED_TYPE.SAME_DAY, label: { id: 'shared.sameDay' } },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.ONE_DAY,
    label: { id: 'shared.dayNumber', values: { number: 1 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.TWO_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 2 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.THREE_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 3 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.SEVEN_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 7 } },
  },
];

export const LISTING_IN_ADVANCE_BOOKING_TYPE = {
  ANY_TIME: 'any_time',
  THREE_MONTHS: '3_months_in_advance',
  SIX_MONTHS: '6_months_in_advance',
  NINE_MONTHS: '9_months_in_advance',
  ONE_YEAR: '1_year',
  UNAVAILABLE: 'unavailable',
};

export const LISTING_IN_ADVANCE_BOOKING_OPTIONS = [
  { value: LISTING_IN_ADVANCE_BOOKING_TYPE.ANY_TIME, label: { id: 'shared.anyTime' } },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.THREE_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 3 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.SIX_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 6 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.NINE_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 9 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.ONE_YEAR,
    label: { id: 'shared.yearNumber', values: { number: 1 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.UNAVAILABLE,
    label: { id: 'shared.datesUnavailableByDefault' },
  },
];

export const LISTING_RESERVATION_PREFERENCES_VALIDATION = {
  GUESTS_STAY_TIME: {
    MIN: 1,
    MAX: 30,
  },
  PREPARATION_TIME: 0,
};

export const LISTING_PREPARATION_TIME_OPTIONS = [
  {
    value: 0,
    label: { id: 'shared.none' },
  },
  {
    value: 1,
    label: { id: 'shared.blockNumberNightBeforeAndAfterEachReservation', values: { number: 1 } },
  },
  {
    value: 2,
    label: { id: 'shared.blockNumberNightBeforeAndAfterEachReservation', values: { number: 2 } },
  },
];

export const LISTING_APPROVAL_POLICY_TYPE = {
  INSTANT_BOOK: 'instant_book',
  TWENTY_FOUR_HOURS_REVIEW: '24_hours_review',
};

export const LISTING_APPROVAL_POLICY_OPTIONS = [
  {
    value: LISTING_APPROVAL_POLICY_TYPE.INSTANT_BOOK,
    label: { id: 'shared.instantBook' },
    description: { id: 'shared.allowsBookingImmediately' },
  },
  {
    value: LISTING_APPROVAL_POLICY_TYPE.TWENTY_FOUR_HOURS_REVIEW,
    label: { id: 'shared.24HourReview' },
    description: { id: 'shared.allows24Hours' },
  },
];

export const LISTING_CANCELLATION_POLICY_TYPE = {
  FLEXIBLE: 'flexible',
  MODERATE: 'moderate',
  STRICT: 'strict',
};

export const LISTING_CANCELLATION_POLICY_OPTIONS = [
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.FLEXIBLE,
    label: { id: 'shared.flexible' },
    description: { id: 'shared.fullRefundNumberDay', values: { number: 1 } },
  },
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.MODERATE,
    label: { id: 'shared.moderate' },
    description: { id: 'shared.fullRefundNumberDay', values: { number: 5 } },
  },
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.STRICT,
    label: { id: 'shared.strict' },
    description: { id: 'shared.percentsRefundUpUntil', values: { percents: '50%' } },
  },
];
