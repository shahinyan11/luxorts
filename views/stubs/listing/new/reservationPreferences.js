import React, { useState } from 'react';
import { Progress, Form, Button, Select, Radio, InputNumber } from 'antd';
import SvgIcon from 'views/stubs/shared/SvgIcon';
import MainHeader from '../../layout/header';

const { Option } = Select;

const ReservationPreferences = () => {
  const [value, setValue] = useState();

  const onChange = (e) => {
    setValue(e.target.value);
  };

  return (
    <div className="main-layout">
      <MainHeader isNewListing newListingText="Reservation preferences" />
      <div className="main-layout__content main-layout__content--full-width">
        <Progress percent={18.75} showInfo={false} className="main-layout__progress" />
        <section className="new-listing pt-32">
          <div className="new-listing__container mb-8">
            <h1 className="new-listing__title mb-32">
              Set reservation preferences for your apartment
            </h1>
            <Form layout="vertical" size="large">
              <h2 className="new-listing__subtitle mb-8">
                When do you need to be notified prior to the arrival of the guests?
              </h2>
              <p className="new-listing__description mb-24">
                At least 1-day notice can help you plan for guests&apos; arrival, but you might miss
                out on last-minute trips.
              </p>
              <div className="new-listing__item mb-24">
                <Radio.Group onChange={onChange} value={value} className="mb-24 d-flex flex-column">
                  <Radio value="same-day">Same day</Radio>
                  <Radio value="1">1 day</Radio>
                  <Radio value="2">2 days</Radio>
                  <Radio value="3">3 days</Radio>
                  <Radio value="7">7 days</Radio>
                </Radio.Group>
              </div>
              <h2 className="new-listing__subtitle mb-8">When can guests check in?</h2>
              <p className="new-listing__description mb-24">
                Select a time range when your guests can check in.
              </p>
              <div className="new-listing__item mb-24">
                <div className="container-two-items">
                  <Form.Item label="From" htmlFor="time-from">
                    <Select
                      placeholder="Select time"
                      id="time-from"
                      suffixIcon={<SvgIcon icon="arrow-down" />}
                    >
                      <Option value="">6:00 AM</Option>
                      <Option value="">7:00 AM</Option>
                      <Option value="">8:00 AM</Option>
                      <Option value="">9:00 AM</Option>
                      <Option value="">10:00 AM</Option>
                      <Option value="">11:00 AM</Option>
                      <Option value="">12:00 PM</Option>
                      <Option value="">1:00 PM</Option>
                      <Option value="">2:00 PM</Option>
                      <Option value="">3:00 PM</Option>
                      <Option value="">4:00 PM</Option>
                      <Option value="">5:00 PM</Option>
                      <Option value="">6:00 PM</Option>
                      <Option value="">7:00 PM</Option>
                      <Option value="">8:00 PM</Option>
                      <Option value="">9:00 PM</Option>
                      <Option value="">10:00 PM</Option>
                      <Option value="">11:00 PM</Option>
                      <Option value="">12:00 AM</Option>
                      <Option value="">1:00 AM</Option>
                      <Option value="">2:00 AM</Option>
                      <Option value="">3:00 AM</Option>
                      <Option value="">4:00 AM</Option>
                      <Option value="">5:00 AM</Option>
                    </Select>
                  </Form.Item>
                  <Form.Item label="To" htmlFor="time-to" className="mb-24">
                    <Select
                      placeholder="Select time"
                      id="time-to"
                      suffixIcon={<SvgIcon icon="arrow-down" />}
                    >
                      <Option value="">6:00 AM</Option>
                      <Option value="">7:00 AM</Option>
                      <Option value="">8:00 AM</Option>
                      <Option value="">9:00 AM</Option>
                      <Option value="">10:00 AM</Option>
                      <Option value="">11:00 AM</Option>
                      <Option value="">12:00 PM</Option>
                      <Option value="">1:00 PM</Option>
                      <Option value="">2:00 PM</Option>
                      <Option value="">3:00 PM</Option>
                      <Option value="">4:00 PM</Option>
                      <Option value="">5:00 PM</Option>
                      <Option value="">6:00 PM</Option>
                      <Option value="">7:00 PM</Option>
                      <Option value="">8:00 PM</Option>
                      <Option value="">9:00 PM</Option>
                      <Option value="">10:00 PM</Option>
                      <Option value="">11:00 PM</Option>
                      <Option value="">12:00 AM</Option>
                      <Option value="">1:00 AM</Option>
                      <Option value="">2:00 AM</Option>
                      <Option value="">3:00 AM</Option>
                      <Option value="">4:00 AM</Option>
                      <Option value="">5:00 AM</Option>
                    </Select>
                  </Form.Item>
                </div>
              </div>
              <h2 className="new-listing__subtitle mb-8">How far in advance can guests book?</h2>
              <p className="new-listing__description mb-24">
                Most hosts keep their calendars updated up to 6 months out.
              </p>
              <div className="new-listing__item mb-24">
                <Radio.Group onChange={onChange} value={value} className="mb-4 d-flex flex-column">
                  <Radio value="any-time">Any time</Radio>
                  <Radio value="3-months">3 months in advance</Radio>
                  <Radio value="6-months">6 months in advance</Radio>
                  <Radio value="9-months">9 months in advance</Radio>
                  <Radio value="1-year">1 year</Radio>
                  <Radio value="unavailable">Dates unavailable by default</Radio>
                </Radio.Group>
                <p className="new-listing__caption pl-32 mb-24">
                  Your entire calendar will be blocked by default, which means you&apos;ll have to
                  manually unblock dates to get booked.
                </p>
              </div>
              <h2 className="new-listing__subtitle mb-8">How long can guests stay?</h2>
              <p className="new-listing__description mb-24">
                Short trips can mean more bookings, but you&apos;ll turn over your space more often.
              </p>
              <div className="new-listing__item mb-24">
                <div className="container-two-items">
                  <Form.Item label="Nights Min" htmlFor="nights-min">
                    <InputNumber
                      min={1}
                      defaultValue={1}
                      id="nights-min"
                      className="ant-input-number-wide"
                      controls={{
                        upIcon: <SvgIcon icon="plus-circle" />,
                        downIcon: <SvgIcon icon="minus-circle" />,
                      }}
                    />
                  </Form.Item>
                  <Form.Item label="Nights Max" htmlFor="nights-max" className="mb-24">
                    <InputNumber
                      min={0}
                      defaultValue={0}
                      id="nights-max"
                      className="ant-input-number-wide"
                      controls={{
                        upIcon: <SvgIcon icon="plus-circle" />,
                        downIcon: <SvgIcon icon="minus-circle" />,
                      }}
                    />
                  </Form.Item>
                </div>
              </div>
              <h2 className="new-listing__subtitle mb-8">
                How much time do you need to prepare your space between reservations?
              </h2>
              <p className="new-listing__description mb-24">
                The availability of your listings largely depends on the length of preparation time
                you need between reservations.
              </p>
              <div className="new-listing__item mb-24">
                <Form.Item label="Preparation Time" htmlFor="preparation-time" className="mb-24">
                  <Select
                    placeholder="None"
                    value="None"
                    id="preparation-time"
                    suffixIcon={<SvgIcon icon="arrow-down" />}
                  >
                    <Option value="">None</Option>
                    <Option value="">Block 1 night before and after each reservation</Option>
                    <Option value="">Block 2 nights before and after each reservation</Option>
                  </Select>
                </Form.Item>
              </div>
              <h2 className="new-listing__subtitle mb-8">Booking approval policy</h2>
              <p className="new-listing__description mb-24">
                Select your default booking approval policy.
              </p>
              <div className="new-listing__item mb-24">
                <Radio.Group
                  onChange={onChange}
                  value={value}
                  className="mb-4 d-flex flex-column ant-radio-group-with-captions"
                >
                  <Radio value="instant-book">Instant book</Radio>
                  <p className="new-listing__description mb-16 pl-32">
                    Allows booking immediately without sending a request for approval.
                  </p>
                  <Radio value="24-hour">24 hour review</Radio>
                  <p className="new-listing__description mb-16 pl-32">
                    Allows 24 hours to accept a booking request.
                  </p>
                </Radio.Group>
              </div>
              <h2 className="new-listing__subtitle mb-8">Cancellation policy</h2>
              <p className="new-listing__description mb-24">
                Select your policy for trip cancellations by guests.
              </p>
              <div className="new-listing__item mb-24">
                <Radio.Group
                  onChange={onChange}
                  value={value}
                  className="mb-4 d-flex flex-column ant-radio-group-with-captions"
                >
                  <Radio value="flexible">Flexible</Radio>
                  <p className="new-listing__description mb-16 pl-32">
                    Full refund 1 day prior to arrival.
                  </p>
                  <Radio value="moderate">Moderate</Radio>
                  <p className="new-listing__description mb-16 pl-32">
                    Full refund 5 days prior to arrival.
                  </p>
                  <Radio value="strict">Strict</Radio>
                  <p className="new-listing__description mb-0 pl-32">
                    50% refund up until 1 week prior to arrival.
                  </p>
                </Radio.Group>
              </div>
            </Form>
          </div>
          <div className="new-listing__footer pt-24 pb-24">
            <div className="new-listing__container d-flex justify-content-space-between">
              <Button htmlType="button" className="main-btn main-btn--tertiary min-width-140">
                Back
              </Button>
              <Button
                htmlType="submit"
                className="main-btn main-btn--primary min-width-140 ml-auto"
              >
                Next
              </Button>
            </div>
          </div>
        </section>
      </div>
    </div>
  );
};

export default ReservationPreferences;
