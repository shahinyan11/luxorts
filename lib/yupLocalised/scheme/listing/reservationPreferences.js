import { equals } from 'ramda';

import yup from 'lib/yupLocalised';
import { LISTING_PRIOR_NOTIFIED_TYPE } from 'constants/listing/reservationPreferences';

const validationSchema = yup.object().shape({
  priorNotified: yup.string().required(),
  priorNotifiedTime: yup.string().when('priorNotified', {
    is: equals(LISTING_PRIOR_NOTIFIED_TYPE.SAME_DAY),
    then: yup.string().nullable().required(),
    otherwise: yup.string().nullable(),
  }),
  checkinStart: yup.string().nullable().required(),
  checkinEnd: yup.string().nullable().required(),
  inAdvanceBooking: yup.string().nullable().required(),
  minNights: yup.number().nullable().required(),
  maxNights: yup.number().nullable().required(),
  approvalPolicy: yup.string().nullable().required(),
  cancellationPolicy: yup.string().nullable().required(),
});

export default validationSchema;
