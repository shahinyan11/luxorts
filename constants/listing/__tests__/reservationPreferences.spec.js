import * as constants from '../reservationPreferences';

it('listing reservation preferences constants match snapshot', () => {
  expect(constants).toMatchSnapshot();
});
