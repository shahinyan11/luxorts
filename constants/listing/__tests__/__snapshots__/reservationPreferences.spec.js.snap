// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listing reservation preferences constants match snapshot 1`] = `
Object {
  "LISTING_APPROVAL_POLICY_OPTIONS": Array [
    Object {
      "description": Object {
        "id": "shared.allowsBookingImmediately",
      },
      "label": Object {
        "id": "shared.instantBook",
      },
      "value": "instant_book",
    },
    Object {
      "description": Object {
        "id": "shared.allows24Hours",
      },
      "label": Object {
        "id": "shared.24HourReview",
      },
      "value": "24_hours_review",
    },
  ],
  "LISTING_APPROVAL_POLICY_TYPE": Object {
    "INSTANT_BOOK": "instant_book",
    "TWENTY_FOUR_HOURS_REVIEW": "24_hours_review",
  },
  "LISTING_CANCELLATION_POLICY_OPTIONS": Array [
    Object {
      "description": Object {
        "id": "shared.fullRefundNumberDay",
        "values": Object {
          "number": 1,
        },
      },
      "label": Object {
        "id": "shared.flexible",
      },
      "value": "flexible",
    },
    Object {
      "description": Object {
        "id": "shared.fullRefundNumberDay",
        "values": Object {
          "number": 5,
        },
      },
      "label": Object {
        "id": "shared.moderate",
      },
      "value": "moderate",
    },
    Object {
      "description": Object {
        "id": "shared.percentsRefundUpUntil",
        "values": Object {
          "percents": "50%",
        },
      },
      "label": Object {
        "id": "shared.strict",
      },
      "value": "strict",
    },
  ],
  "LISTING_CANCELLATION_POLICY_TYPE": Object {
    "FLEXIBLE": "flexible",
    "MODERATE": "moderate",
    "STRICT": "strict",
  },
  "LISTING_IN_ADVANCE_BOOKING_OPTIONS": Array [
    Object {
      "label": Object {
        "id": "shared.anyTime",
      },
      "value": "any_time",
    },
    Object {
      "label": Object {
        "id": "shared.numberMonthsInAdvance",
        "values": Object {
          "number": 3,
        },
      },
      "value": "3_months_in_advance",
    },
    Object {
      "label": Object {
        "id": "shared.numberMonthsInAdvance",
        "values": Object {
          "number": 6,
        },
      },
      "value": "6_months_in_advance",
    },
    Object {
      "label": Object {
        "id": "shared.numberMonthsInAdvance",
        "values": Object {
          "number": 9,
        },
      },
      "value": "9_months_in_advance",
    },
    Object {
      "label": Object {
        "id": "shared.yearNumber",
        "values": Object {
          "number": 1,
        },
      },
      "value": "1_year",
    },
    Object {
      "label": Object {
        "id": "shared.datesUnavailableByDefault",
      },
      "value": "unavailable",
    },
  ],
  "LISTING_IN_ADVANCE_BOOKING_TYPE": Object {
    "ANY_TIME": "any_time",
    "NINE_MONTHS": "9_months_in_advance",
    "ONE_YEAR": "1_year",
    "SIX_MONTHS": "6_months_in_advance",
    "THREE_MONTHS": "3_months_in_advance",
    "UNAVAILABLE": "unavailable",
  },
  "LISTING_PREPARATION_TIME_OPTIONS": Array [
    Object {
      "label": Object {
        "id": "shared.none",
      },
      "value": 0,
    },
    Object {
      "label": Object {
        "id": "shared.blockNumberNightBeforeAndAfterEachReservation",
        "values": Object {
          "number": 1,
        },
      },
      "value": 1,
    },
    Object {
      "label": Object {
        "id": "shared.blockNumberNightBeforeAndAfterEachReservation",
        "values": Object {
          "number": 2,
        },
      },
      "value": 2,
    },
  ],
  "LISTING_PRIOR_NOTIFIED_OPTIONS": Array [
    Object {
      "label": Object {
        "id": "shared.sameDay",
      },
      "value": "same_day",
    },
    Object {
      "label": Object {
        "id": "shared.dayNumber",
        "values": Object {
          "number": 1,
        },
      },
      "value": "1_day",
    },
    Object {
      "label": Object {
        "id": "shared.dayNumberPlural",
        "values": Object {
          "number": 2,
        },
      },
      "value": "2_days",
    },
    Object {
      "label": Object {
        "id": "shared.dayNumberPlural",
        "values": Object {
          "number": 3,
        },
      },
      "value": "3_days",
    },
    Object {
      "label": Object {
        "id": "shared.dayNumberPlural",
        "values": Object {
          "number": 7,
        },
      },
      "value": "7_days",
    },
  ],
  "LISTING_PRIOR_NOTIFIED_TYPE": Object {
    "ONE_DAY": "1_day",
    "SAME_DAY": "same_day",
    "SEVEN_DAYS": "7_days",
    "THREE_DAYS": "3_days",
    "TWO_DAYS": "2_days",
  },
  "LISTING_RESERVATION_PREFERENCES_VALIDATION": Object {
    "GUESTS_STAY_TIME": Object {
      "MAX": 30,
      "MIN": 1,
    },
    "PREPARATION_TIME": 0,
  },
}
`;
